<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1071</width>
    <height>790</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="statusTip">
   <string>test</string>
  </property>
  <property name="documentMode">
   <bool>false</bool>
  </property>
  <property name="unifiedTitleAndToolBarOnMac">
   <bool>false</bool>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="0" column="0">
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QOpenCvImageBox" name="cameraWidget" native="true"/>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1071</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QDockWidget" name="dockWidget">
   <property name="minimumSize">
    <size>
     <width>326</width>
     <height>282</height>
    </size>
   </property>
   <property name="allowedAreas">
    <set>Qt::LeftDockWidgetArea|Qt::RightDockWidgetArea</set>
   </property>
   <property name="windowTitle">
    <string>Object detection</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <item>
      <widget class="QCheckBox" name="faceDetectionEnabled">
       <property name="text">
        <string>Enabled</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox">
       <property name="title">
        <string>Options</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout">
        <property name="margin">
         <number>5</number>
        </property>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <item>
           <widget class="QSpinBox" name="imageScaleSB">
            <property name="prefix">
             <string>Image scale: </string>
            </property>
            <property name="singleStep">
             <number>1</number>
            </property>
            <property name="value">
             <number>8</number>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QComboBox" name="cascadesCB">
          <item>
           <property name="text">
            <string>haarcascade_frontalface_alt2.xml</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>haarcascade_frontalface_default.xml</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>haarcascade_frontalface_alt.xml</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>haarcascade_frontalface_alt_tree.xml</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>haarcascade_eye.xml</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>haarcascade_eye2.xml</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>haarcascade_eye_tree_eyeglasses.xml</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>haarcascade_lefteye_2splits.xml</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>haarcascade_fullbody.xml</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>haarcascade_lowerbody.xml</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>haarcascade_mcs_eyepair_big.xml</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>haarcascade_mcs_eyepair_small.xml</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>haarcascade_mcs_lefteye.xml</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>haarcascade_mcs_mouth.xml</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>haarcascade_mcs_nose.xml</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>haarcascade_mcs_righteye.xml</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>haarcascade_mcs_upperbody.xml</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>haarcascade_profileface.xml</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>haarcascade_righteye_2splits.xml</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>haarcascade_upperbody.xml</string>
           </property>
          </item>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="faceDetectionGB">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="title">
        <string/>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_7">
        <property name="margin">
         <number>5</number>
        </property>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QLabel" name="label">
            <property name="text">
             <string>Objs found: </string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="objsFoundLabel">
            <property name="text">
             <string>0</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QLabel" name="calcTimeLabel">
          <property name="text">
           <string>Calc time: 0</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <spacer name="verticalSpacer">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeType">
        <enum>QSizePolicy::Expanding</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>40</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockWidget_2">
   <property name="minimumSize">
    <size>
     <width>198</width>
     <height>223</height>
    </size>
   </property>
   <property name="floating">
    <bool>false</bool>
   </property>
   <property name="allowedAreas">
    <set>Qt::LeftDockWidgetArea|Qt::RightDockWidgetArea</set>
   </property>
   <property name="windowTitle">
    <string>HSV</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_2">
    <layout class="QVBoxLayout" name="verticalLayout_4">
     <item>
      <widget class="QCheckBox" name="hsvAllTheImage">
       <property name="text">
        <string>All the image</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_4">
       <property name="title">
        <string>Values</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_8">
        <item>
         <widget class="QSpinBox" name="hsvHueSB">
          <property name="prefix">
           <string>Hue interval: </string>
          </property>
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>255</number>
          </property>
          <property name="singleStep">
           <number>1</number>
          </property>
          <property name="value">
           <number>43</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="hsvBrightnessThresholdSB">
          <property name="suffix">
           <string/>
          </property>
          <property name="prefix">
           <string>Brightness threshold: </string>
          </property>
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>255</number>
          </property>
          <property name="value">
           <number>56</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="hsvSaturationThresholdSB">
          <property name="prefix">
           <string>Saturation threshold: </string>
          </property>
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>255</number>
          </property>
          <property name="value">
           <number>20</number>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <spacer name="verticalSpacer_2">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>40</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QDockWidget" name="dockWidget_3">
   <property name="windowTitle">
    <string>Source</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_3">
    <layout class="QVBoxLayout" name="verticalLayout_5">
     <item>
      <widget class="QGroupBox" name="groupBox_2">
       <property name="title">
        <string>Time out new frame</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_6">
        <item>
         <widget class="QSpinBox" name="timeOutCamera">
          <property name="suffix">
           <string> ms</string>
          </property>
          <property name="prefix">
           <string/>
          </property>
          <property name="minimum">
           <number>10</number>
          </property>
          <property name="maximum">
           <number>2000</number>
          </property>
          <property name="singleStep">
           <number>50</number>
          </property>
          <property name="value">
           <number>100</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton">
          <property name="text">
           <string>Set time out</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_3">
       <property name="title">
        <string>Info</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QLabel" name="imageSizeInfoLabel">
          <property name="text">
           <string>Image size:</string>
          </property>
          <property name="buddy">
           <cstring></cstring>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <spacer name="verticalSpacer_3">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>29</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </widget>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QOpenCvImageBox</class>
   <extends>QWidget</extends>
   <header>qopencvimagebox.h</header>
   <container>1</container>
   <slots>
    <slot>setHsvAllImageConvertedEnabled(bool)</slot>
    <slot>setDetectionEnabled(bool)</slot>
    <slot>setODImageScale(int)</slot>
    <slot>setCascadeName(QString)</slot>
    <slot>setHsvHueInterval(int)</slot>
    <slot>setHsvBrightnessThreshold(int)</slot>
    <slot>setHsvSaturationThreshold(int)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>hsvAllTheImage</sender>
   <signal>toggled(bool)</signal>
   <receiver>cameraWidget</receiver>
   <slot>setHsvAllImageConvertedEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>85</x>
     <y>359</y>
    </hint>
    <hint type="destinationlabel">
     <x>650</x>
     <y>413</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>faceDetectionEnabled</sender>
   <signal>toggled(bool)</signal>
   <receiver>faceDetectionGB</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>44</x>
     <y>67</y>
    </hint>
    <hint type="destinationlabel">
     <x>84</x>
     <y>185</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>faceDetectionEnabled</sender>
   <signal>toggled(bool)</signal>
   <receiver>cameraWidget</receiver>
   <slot>setDetectionEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>19</x>
     <y>66</y>
    </hint>
    <hint type="destinationlabel">
     <x>477</x>
     <y>204</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>imageScaleSB</sender>
   <signal>valueChanged(int)</signal>
   <receiver>cameraWidget</receiver>
   <slot>setODImageScale(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>177</x>
     <y>110</y>
    </hint>
    <hint type="destinationlabel">
     <x>357</x>
     <y>155</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cascadesCB</sender>
   <signal>currentIndexChanged(QString)</signal>
   <receiver>cameraWidget</receiver>
   <slot>setCascadeName(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>141</x>
     <y>148</y>
    </hint>
    <hint type="destinationlabel">
     <x>665</x>
     <y>231</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hsvHueSB</sender>
   <signal>valueChanged(int)</signal>
   <receiver>cameraWidget</receiver>
   <slot>setHsvHueInterval(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>169</x>
     <y>398</y>
    </hint>
    <hint type="destinationlabel">
     <x>429</x>
     <y>378</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hsvBrightnessThresholdSB</sender>
   <signal>valueChanged(int)</signal>
   <receiver>cameraWidget</receiver>
   <slot>setHsvBrightnessThreshold(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>141</x>
     <y>440</y>
    </hint>
    <hint type="destinationlabel">
     <x>411</x>
     <y>432</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hsvSaturationThresholdSB</sender>
   <signal>valueChanged(int)</signal>
   <receiver>cameraWidget</receiver>
   <slot>setHsvSaturationThreshold(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>240</x>
     <y>479</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>485</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
